packages/trader/src/Stores/Modules/Trading/trade-store.js

```diff
@@ -421,7 +423,19 @@ export default class TradeStore extends BaseStore {
                         type: response.msg_type,
                         ...response.error,
                     };
-                    this.root_store.ui.toggleServicesErrorModal(true);
+                    
+                    // check if token is invalid
+                    if (response.error.code === 'InvalidToken') {
+                    // send logout, clear contracts, clean up client and then show message
+                        this.root_store.client.cleanUp();
+                        requestLogout().then(() => {
+                            this.root_store.modules.trade.clearContracts();
+                            this.root_store.ui.toggleServicesErrorModal(true);
+                        });
+                    } else {
+                        this.root_store.ui.toggleServicesErrorModal(true); // for other scenarios
```

Comment:
imo, it will work if you put `this.root_store.ui.toggleServicesErrorModal(true);` outside if else, bcause both of conditions will toggle the modal, and the logout will happen in the background

Reply:
@mustofa-binary I tried that but when we call clean up it will clear message. The async behavior of some actions in cleanup caused this, so we need some delay to call toggleServicesErrorModal.